drp.py

```diff
@@ -30,55 +30,42 @@ def base_activity():
         activity['timestamps'] = {'start': start_time}
     return activity
 
-
-# TODO base these on base scope name
-EXT_ICON_MAP = {
-    '.js': 'javascript',
-    '.py': 'python',
-    '.lua': 'lua',
-    '.rb': 'ruby',
-    '.gemspec': 'ruby',
-    '.cr': 'crystal',
-    '.css': 'css',
-    '.html': 'html',
-    '.htm': 'html',
-    '.shtml': 'html',
-    '.xhtml': 'html',
-    '.md': 'markdown',
-    '.mdown': 'markdown',
-    '.markdown': 'markdown',
-    '.markdn': 'markdown',
-    '.cs': 'cs',
-    '.csproj': 'cs',
-    '.cpp': 'cpp',
-    '.php': 'php',
-    '.php3': 'php',
-    '.go': 'go',
-    '.d': 'd',
-    '.c': 'c',
-    '.json': 'json',
-    '.exs': 'elixir',
-    '.ex': 'elixir',
-    '.java': 'java',
-    '.properties': 'java',
-    '.ts': 'typescript',
-    '.sublime-settings': 'json',
-    '.sublime-snippet': 'json',
-    '.sublime-theme': 'json',
-    '.sublime-menu': 'json',
-    '.sublime-commands': 'json',
-    '.sublime-keymap': 'json',
-    '.sublime-mousemap': 'json',
-    '.sublime-build': 'json',
-    '.sublime-macro': 'json',
-    '.sublime-completions': 'json',
-    '.sublime-project': 'json'
+SCOPE_ICON_MAP = {
+    'source.c': 'c',
```

Comment:
Many of these map to the same scope as would be used by default anyway (the second segment).

I think it's better to define a list of supported icon identifiers (the ones that have been uploaded) and if the resolved icon name is not within that list, fall back to `unknown`. In addition to the map for special associations, e.g. `text.html.markdown -> markdown`, of course.

Reply:
I split the list of icons in two.
The icons that have the same name that the language are in the set AVAILABLES_ICONS
The ones that have a different name (like 'javascript' or that use a more specific icon (like 'text.html.markdown') are in SCOPE_ICON_MAP.

Note that C++, JS and TS icons could be rename to match their language name.